src/test/java/com/amazon/opendistroforelasticsearch/security/user/AuthCredentialsTests.java

```diff
@@ -0,0 +1,19 @@
+package com.amazon.opendistroforelasticsearch.security.user;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+public class AuthCredentialsTests {
+  @Test
+  public void testEquality() {
```

Comment:
Should we add 2 test cases?
1. `object.internalPasswordHash == null` && `other. internalPasswordHash == null` ==> equals() returns true
2. `object.internalPasswordHash == null` && `other. internalPasswordHash == <anything other than null>` returns false.

Reply:
@nicot Can you add these 2 test cases?

Reply:
A user password will be null if we set the `backendRoles`.

> object.internalPasswordHash == null && other.internalPasswordHash == null ==> equals() returns true

This is the test case on line 9 — the users have backendRoles, so the password is null for both.


> object.internalPasswordHash == null && other.internalPasswordHash == <non-null value> ==> equals() returns false.

This is the test case on line 28 — one user has backendRoles and one has a password, and they are not equal.